add Zip/Repeat typeclass with instances (#1814)
arrow-kt/arrow: modules/docs/arrow-docs/docs/docs/arrow/typeclasses/repeat/README.md

---
layout: docs
title: Repeat
permalink: /docs/arrow/typeclasses/repeat/
redirect_from:
  - /docs/typeclasses/repeat/
---

> ## Zip

Reviewer: ```suggestion
## Repeat
```

Reviewer: +

{:.beginner}
beginner

> The `Repeat` typeclass extends the `Zip` typeclass with an repeat structure.

Reviewer: ```suggestion
The `Repeat` typeclass extends the `Zip` typeclass with a way to repeat the structure.
```
The previous sounded like `Repeat` adds any capability to the datatype itself when it actually just provides methods for that datatype

Reviewer: +

### Main Combinators

#### repeat

> Provides an structure that can be used to zip with. The structure contains the provided value and is

Reviewer: ```suggestion
Provides a structure that can be used to zip with. The structure repeats the provided value and is
```
I think repeats describes this a bit better

Reviewer: +

potentially infinite.

`fun <A> repeat(a: A): Kind<F, A>`

```kotlin:ank
import arrow.core.extensions.*
import arrow.core.extensions.sequencek.repeat.repeat
import arrow.core.*

val seq = generateSequence(0) { it + 1 }.k()
SequenceK.repeat().run {
    repeat("Item").zip(seq).fix().take(5).toList()
}
```

### Laws

Arrow provides [`RepeatLaws`][functor_laws_source]{:target="_blank"} in the form of test cases for internal verification of lawful instances and third party apps creating their own Repeat instances.

#### Creating your own `Repeat` instances

> Arrow already provides Align instances for common datatypes (e.g. Option, SequenceK). See their implementations

Reviewer: ```suggestion
Arrow already provides Repeat instances for common datatypes (e.g. Option, SequenceK). See their implementations
```

Reviewer: +
